lib/ansible/plugins/inventory/docker_swarm.py

```diff
@@ -0,0 +1,80 @@
+
+from __future__ import (absolute_import, division, print_function)
+
+__metaclass__ = type
+
+DOCUMENTATION = '''
+    name: docker_swarm
+    plugin_type: inventory
+    authors:
+      - Stefan Heitmüller (stefan.heitmueller@gmx.com)
+    short_description: Ansible dynamic inventory plugin for Docker swarm nodes.
+    requirements:
+        - python >= 2.7
+        - Docker SDK for Python
+    description:
+        - Reads inventories from the Docker swarm API.
+        - Uses a YAML configuration file docker_swarm.[yml|yaml].
+    options:
+        plugin:
+            description: the name of this plugin, it should always be set to 'docker_swarm' for this plugin to recognize it as it's own.
+            required: true
+            choices: ['docker_swarm']
+        host:
+            description: Socket of a Docker swarm manager node (tcp,unix).
+            default: 'unix://var/run/docker.sock'
+'''
+
+EXAMPLES = '''
+# docker_swarm.yml
+plugin: docker_swarm
+host: unix:///var/run/docker.sock
+'''
+
+from ansible.errors import AnsibleError, AnsibleParserError
+from ansible.module_utils._text import to_native
+from ansible.plugins.inventory import BaseInventoryPlugin
+from ansible.release import __version__
+
+try:
+    import docker
+except ImportError:
+    raise AnsibleError('The Docker swarm dynamic inventory plugin requires the docker python binding: https://github.com/docker/docker-py.')
+
+
+class InventoryModule(BaseInventoryPlugin):
+    ''' Host inventory parser for ansible using Docker swarm as source. '''
+
+    NAME = "docker_swarm"
+
+    def _populate(self):
+        client = docker.DockerClient(base_url=self.get_option('host'))
+        self.inventory.add_group('manager')
+        self.inventory.add_group('worker')
+        try:
+            manager_nodes = client.nodes.list(filters={'role': 'manager'})
+            worker_nodes = client.nodes.list(filters={'role': 'worker'})
+        except ConnectionError:
+            raise AnsibleError('Unable to connect to the Docker swarm API.')
+        for node in manager_nodes:
+            node_hostname = client.nodes.get(node.id).attrs['Description']['Hostname']
+            self.inventory.add_host(node_hostname)
+            self.inventory.add_host(node_hostname, group='manager')
+        for node in worker_nodes:
+            node_hostname = client.nodes.get(node.id).attrs['Description']['Hostname']
+            self.inventory.add_host(node_hostname)
+            self.inventory.add_host(node_hostname, group='worker')
+
+    def verify_file(self, path):
+        """Return the possibly of a file being consumable by this plugin."""
+        return (
+            super(InventoryModule, self).verify_file(path) and
+            path.endswith((self.NAME + ".yaml", self.NAME + ".yml")))
+
+    def parse(self, inventory, loader, path, cache=True):
+        super(InventoryModule, self).parse(inventory, loader, path, cache)
+        self._read_config_data(path)
```

Comment:
What information are included in a configuration file?

Reply:
See the `EXAMPLES` section. For now it is `host` only, but theres no support for TLS secured Docker sockets right now, so this would be extended to include certs and keys. Probably a filter later?

Reply:
TLS is a must be, I cannot test the plugin without it as all my Swarms run on TLS with certs signed by own CA. I think the options for TLS should be the same as for al `docker` modules.

Reply:
We can add this in a second PR. @WojciechowskiPiotr you could also keep this as a branch based on @morph027's with your changes as a new commit, then you can pull and rebase whenever @morph027 adds something, and later (once this PR is merged) create a new PR from your branch. Except, of course, if @morph027 already wants to include TLS support.

Reply:
Jip, would think so to. I've started to tinker with TLS based on @WojciechowskiPiotr code, but this adds a lot of complexity, so for now i'd like to have the basic plugin and it's structure ready and improve afterwards based on it.